python/mxnet/gluon/block.py

```diff
@@ -1094,8 +1094,57 @@ def _deferred_infer_shape(self, *args):
             raise ValueError(error_msg)
 
     def _call_cached_op(self, *args):
-        if self._cached_op is None:
+        if not self._cached_op_args or not self._cached_graph:
             self._build_cache(*args)
+
+        if self._first_forward:
+            # partition static shape ops if the graph contains any dynamic shape op
+            self._first_forward = False
+            data, out = self._cached_graph
+            out, is_dynamic = out._optimize_for_dynamic_shape_op(is_np_array(), self._flags)
```

Comment:
are you still planning on replacing this with `optimize_for` instead of creating your own Python API binding?

Reply:
Yes, done!